doozer/doozerlib/cli/olm_bundle.py

```diff
@@ -201,42 +211,107 @@ def rebase_and_build(olm_bundle: OLMBundle):
+            LOGGER.error('Error during rebase or build for: {}'.format(operator_nvr))
             record['message'] = str(err)
+
         finally:
             runtime.record_logger.add_record("build_olm_bundle", **record)
             return record
 
+    def update_konflux_db(olm_bundle, record, build_info):
+        if dry_run:
+            return
+
+        if not runtime.konflux_db:
+            LOGGER.warning('Konflux DB connection is not initialized, not writing build record to the Konflux DB.')
+            return
+
+        runtime.konflux_db.bind(KonfluxBundleBuildRecord)
+        try:
+            dfp = DockerfileParser(f'{olm_bundle.bundle_clone_path}/Dockerfile')
+            source_repo, commitish = dfp.labels['io.openshift.build.commit.url'].split('/commit/')
+            _, rebase_repo_url, _ = gather_git(['-C', olm_bundle.bundle_clone_path, 'remote', 'get-url', 'origin'])
+            _, rebase_commitish, _ = gather_git(['-C', olm_bundle.bundle_clone_path, 'rev-parse', 'HEAD'])
+
+            build_record_params = {
+                'name': olm_bundle.bundle_name,
+                'group': runtime.group,
+                'assembly': runtime.assembly,
+                'source_repo': source_repo,
+                'commitish': commitish,
+                'rebase_repo_url': convert_remote_git_to_https(rebase_repo_url),
+                'rebase_commitish': rebase_commitish.strip(),
+                'artifact_type': ArtifactType.IMAGE,
+                'engine': Engine.BREW,
+                'art_job_url': os.getenv('BUILD_URL', 'n/a'),
+                'build_pipeline_url': record['task_url'],
+                'pipeline_commit': 'n/a',
+                'operator_nvr': olm_bundle.operator_dict['nvr']
+            }
+
+            if record['status'] == 0:
+                image_pullspec = build_info['extra']['image']['index']['pull'][0]
+
+                build_record_params.update({
+                    'version': build_info['version'],
+                    'release': build_info['release'],
+                    'start_time': datetime.strptime(build_info['start_time'], '%Y-%m-%d %H:%M:%S.%f'),
+                    'end_time': datetime.strptime(build_info['completion_time'], '%Y-%m-%d %H:%M:%S.%f'),
+                    'image_pullspec': image_pullspec,
+                    'image_tag': image_pullspec.split('sha256:')[-1],
+                    'operand_nvrs': [v for _, v in olm_bundle.found_image_references.items()],
+                    'build_id': str(build_info['id']),
+                    'outcome': KonfluxBuildOutcome.SUCCESS,
+                    'nvr': record['bundle_nvr']
+                })
+
+            else:
+                build_record_params.update({
+                    'outcome': KonfluxBuildOutcome.FAILURE,
+                    'start_time': datetime.now(),  # placeholder, cannot be NULL
+                    'end_time': datetime.now(),  # placeholder, cannot be NULL
```

Comment:
I'm thinking build_info values would still be usable here?

Reply:
In case of failures, Brew does not return these timestamps AFAICS. We might consider storing the task related timestamps, if possible

Reply:
right, i think we should store task id / task info, it would be useful for art-dash as well

Reply:
I'll note down a follow up to this PR, so I can merge (and move on, I need this refactoring for other subtasks)